src/server.cpp

```diff
@@ -421,10 +421,24 @@ CServer::CServer ( const int          iNewMaxNumChan,
         vecChannels[i].SetEnable ( true );
     }
 
-    // introduced by kraney (#653): "increased the size of the thread pool"
-    if ( bUseMultithreading )
-    {
-        QThreadPool::globalInstance()->setMaxThreadCount ( QThread::idealThreadCount() * 4 );
+    int iAvailableCores = QThread::idealThreadCount();
+
+    // setup QThreadPool if multithreading is active and possible
+    if ( bUseMultithreading ) {
+        if ( iAvailableCores == 1 )
+        {
+            qDebug() << "found only one core, disabling multithreading";
+            bUseMultithreading = false;
+        }
+        else
+        {
+            // set maximum thread count to available cores; other threads will share at random
+            iMaxNumThreads = iAvailableCores;
```

Comment:
Should this be configurable, btw?
If not all cores are usuable by this Jamulus instance, more threads will just add more overhead, I'd assume?

See also https://github.com/jamulussoftware/jamulus/pull/960#issuecomment-782758759

The previous multithreading logic was not configurable either, so this should not be a blocker for this PR, I guess.

Reply:
I've had a think about this, and I don't think reserving cores actually gains us anything. Certainly there seems no point in dividing the audio processing across more threads than there are cores. But using fewer threads would mean it is usually not using cores optimally. Using all the cores means they will have approximately equal amounts of spare capacity, which can be used by other lighter threads or by other programs, even other Jamulus instances. If there was, say, an additional Jamulus instance, it too could make use of the spare capacity on all the cores. If we "reserved" a core by using fewer threads, I don't think there is any way for the second instance to specifically use the less busy core instance (I suspect it just uses any free core, but this is way deeper than I know), so we may as well allow the free time to be distributed evenly.

Reply:
I think it's possible (in Linux, anyway), to limit the number of cores visible to a process, so it's not really needed in the code itself.  (You can also essentially tell a process to use a specific core by only letting it see that core.)